graphql/graphql.go

```diff
@@ -39,6 +41,27 @@ var (
 	errBlockInvariant = errors.New("block objects must be instantiated with at least one of num or hash")
 )
 
+type Long uint64
+
+// ImplementsGraphQLType returns true if Long implements the provided GraphQL type.
+func (b Long) ImplementsGraphQLType(name string) bool { return name == "Long" }
+
+// UnmarshalGraphQL unmarshals the provided GraphQL query data.
+func (b *Long) UnmarshalGraphQL(input interface{}) error {
+	var err error
+	switch input := input.(type) {
+	case string:
+		value, err := strconv.ParseUint(input, 10, 64)
+		*b = Long(value)
+		return err
```

Comment:
This feels a bit breaking. Generally, when interacting with geth, you can feed it strings representing integers. In all those cases, the string is `0x`-prefixed hex encodings. Now, in this particular instance, you have to feed it `"10"`  instead of `"0x0a"`. 

So basically, some options are :
- Allow `0x`-prefixed hex-encoded strings on input, and non-`0x`-prefixed decimal encoded strings, 
- Only allow non-`0x`-prefixed decimal encoded strings (the way it is in this PR)
- Reject string-encoded values

I'm not sure which is best, just wanted to raise the topic.

Reply:
Hm, actually, I don't quite get this. 
If I modify the testcase like so: 
```
	body := strings.NewReader(`{"query": "{block(number:\"0\"){number,gasUsed,gasLimit}}","variables": null}`)
```
Then I get 
```
        	            	expected: "{\"data\":{\"block\":{\"number\":0,\"gasUsed\":0,\"gasLimit\":11500000}}}"
        	            	actual  : "{\"errors\":[{\"message\":\"hex string without 0x prefix\"}],\"data\":{}}"
```
So it appears that block number selectors on input are still required to be 0x-prefixed (if strings).

Reply:
That's a bug. I'll fix it now.

The schema of the EIP is clear: decimals only. You can probably support strings for hex values but that's beyond what the spec recommends and won't be in line with other clients.

Reply:
:+1:  ambiguity == bad, strictness == good